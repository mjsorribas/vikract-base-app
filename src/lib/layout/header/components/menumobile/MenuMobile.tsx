import {
  Box,
  Flex,
  Spacer,
  Link,
  MenuItem,
  MenuList,
  MenuButton,
  Menu,
  IconButton,
  MenuDivider,
  useDisclosure,
} from "@chakra-ui/react";
import { RiAlignJustify, RiCloseFill } from "react-icons/ri";

import MenuAccess from "../menuaccess/menuAccess";
import Logo from "lib/components/logo/Logo";
import { theme } from "lib/styles/customTheme";

export default function MenuMobile() {
  const { isOpen, onToggle } = useDisclosure();
  return (
    <Flex>
      <Logo boxSize="25px" fontSize="1.168rem" fontColor={theme.colors.black} />
      <Spacer />
      <Box display="flex" alignItems="center">
        <Menu>
          <MenuButton
            onClick={onToggle}
            as={IconButton}
            size="lg"
            border="none"
            aria-label="Options"
            icon={isOpen ? <RiCloseFill /> : <RiAlignJustify />}
            variant="outline"
          />
          <MenuList
            w="100vw"
            alignContent="center"
            backgroundColor={theme.colors.cream}
          >
            <MenuItem
              w="100%"
              h="100%"
              display="flex"
              alignContent="center"
              alignItems="center"
            >
              <Link
                w="100%"
                h="100%"
                fontWeight="bold"
                mt={2}
                mb={2}
                textAlign="center"
                _hover={{ textDecoration: "none" }}
                href="/"
              >
                Home
              </Link>
            </MenuItem>
            <MenuItem>
              <Link
                w="100%"
                mt={2}
                mb={2}
                textAlign="center"
                _hover={{ textDecoration: "none" }}
                href="/aboutus"
              >
                About Us
              </Link>
            </MenuItem>
            <MenuItem>
              <Link
                w="100%"
                mt={2}
                mb={2}
                textAlign="center"
                _hover={{ textDecoration: "none" }}
                href="/contactus"
              >
                Contact Us
              </Link>
            </MenuItem>
            <MenuDivider />
            <MenuAccess />
          </MenuList>
        </Menu>
      </Box>
    </Flex>
  );
}
