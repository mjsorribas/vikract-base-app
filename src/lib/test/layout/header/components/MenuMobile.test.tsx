import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { expect, test } from "vitest";

import MenuMobile from "lib/layout/header/components/menumobile/MenuMobile";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
};

test("MenuMobile", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: "/" }]}>
      <MenuMobile />
    </MemoryRouter>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
