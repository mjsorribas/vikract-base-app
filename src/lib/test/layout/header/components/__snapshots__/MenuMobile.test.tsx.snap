// Vitest Snapshot v1

exports[`MenuMobile 1`] = `
<div
  className="css-k008qs"
>
  <div
    className="css-70qvj9"
  >
    <div
      className="css-0"
    >
      <img
        className="chakra-image css-1chmwhm"
        src="/assets/vikract.svg"
        title="The Vikract Webapp template"
      />
      <p
        className="chakra-text css-104caxi"
      >
        THE VIKRACT TEMPLATE
      </p>
    </div>
  </div>
  <div
    className="css-17xejub"
  />
  <div
    className="css-70qvj9"
  >
    <button
      aria-controls="menu-list-:r2:"
      aria-expanded={false}
      aria-haspopup="menu"
      aria-label="Options"
      className="chakra-button chakra-menu__menu-button css-ymfqft"
      id="menu-button-:r2:"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M3 4h18v2H3V4zm0 15h18v2H3v-2zm0-5h18v2H3v-2zm0-5h18v2H3V9z"
          />
        </g>
      </svg>
    </button>
    <div
      className="css-0"
      style={
        {
          "inset": "0 auto auto 0",
          "minWidth": "max-content",
          "position": "absolute",
          "visibility": "hidden",
        }
      }
    >
      <div
        aria-orientation="vertical"
        className="chakra-menu__menu-list css-luqtzj"
        id="menu-list-:r2:"
        onKeyDown={[Function]}
        role="menu"
        style={
          {
            "opacity": 0,
            "transform": "scale(0.8) translateZ(0)",
            "transformOrigin": "var(--popper-transform-origin)",
            "visibility": "hidden",
          }
        }
        tabIndex={-1}
      >
        <button
          className="chakra-menu__menuitem css-115esnj"
          id="menu-list-:r2:-menuitem-:r3:"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          role="menuitem"
          tabIndex={0}
          type="button"
        >
          <a
            className="chakra-link css-f9i6ey"
            href="/"
          >
            Home
          </a>
        </button>
        <button
          className="chakra-menu__menuitem css-1k756t9"
          id="menu-list-:r2:-menuitem-:r4:"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          role="menuitem"
          tabIndex={0}
          type="button"
        >
          <a
            className="chakra-link css-1oxylfh"
            href="/prices"
          >
            Prices
          </a>
        </button>
        <button
          className="chakra-menu__menuitem css-1k756t9"
          id="menu-list-:r2:-menuitem-:r5:"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          role="menuitem"
          tabIndex={0}
          type="button"
        >
          <a
            className="chakra-link css-1oxylfh"
            href="/contactus"
          >
            Contact Us
          </a>
        </button>
        <hr
          aria-orientation="horizontal"
          className="chakra-menu__divider css-0"
        />
        <div
          className="css-0"
        >
          <div
            className="css-0"
          >
            <div
              className="chakra-stack css-111rh5x"
            >
              <button
                className="chakra-button css-4xx2wk"
                type="button"
              >
                Sign Up
              </button>
              <button
                className="chakra-button css-4xx2wk"
                type="button"
              >
                Sign In
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
